pkgs/by-name/na/navidrome/plugins/main-repo-plugin-builder.nix

```
@@ -0,0 +1,31 @@
+{
+  name,
+  buildGo124Module,
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  navidrome,
+}:
+
+# Builds plugins from the navidrome example folder
+navidrome.buildGoModule rec {
+  pname = name;
+  inherit (navidrome) src version vendorHash;
+
+  buildPhase = ''
+    runHook preBuild
+
+    GOOS=wasip1 GOARCH=wasm go build -buildmode=c-shared -o plugin.wasm ./plugins/examples/${pname}/...
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    DEST=$out/share/plugins/${pname}
+    install -m 755 -Dt $out/share/${pname} plugin.wasm plugins/examples/${pname}/manifest.json
```

```suggestion
    GOOS=wasip1 GOARCH=wasm go build -buildmode=c-shared -o plugin.wasm ./plugins/examples/${name}/...

    runHook postBuild
  '';

  installPhase = ''
    runHook preInstall

    DEST=$out/share/plugins/${name}
    install -m 755 -Dt $out/share/${name} plugin.wasm plugins/examples/${name}/manifest.json
```
Using pname like this is an anti feature but since we have a legitimate usecase here, we should probably use the input variable name instead.